README.adoc

```diff
@@ -38,153 +40,199 @@ endif::[]
 
-[source,bash]
-----
-sdk install java 8.0.265-amzn
-sdk install kotlin 1.4.10
-----
+===== Setup Maven Repository
 
-Check your installation by:
+Add the following maven repository configuration to your root build.gradle:
 
-[source,bash]
+[source, Gradle]
 ----
-java -version
-kotlin -version
+    allprojects {
+        repositories {
+            ...
+            maven {
+                url = URI("https://maven.pkg.github.com/d4l-data4life/hc-fhir-sdk-kmp")
+                credentials {
+                    username = project.findProperty("gpr.user") as String? ?: System.getenv("PACKAGE_REGISTRY_USERNAME")
+                    password = project.findProperty("gpr.key") as String? ?: System.getenv("PACKAGE_REGISTRY_TOKEN")
+                }
+            }
+        }
+    }
 ----
 
-==== Install Python
+== Usage
 
-It is recommended to use link:https://github.com/pyenv/pyenv[PyEnv] for managing dedicated python versions.
+How to use this SDK.
 
-Use link:https://brew.sh/[Homebrew] to install it on mac:
+=== Serialization
 
-[source,bash]
-----
-brew install pyenv
+Create an instance of your desired `FhirStu3Parser` or `FhirR4Parser` using `FhirParserFactory.createStu3Parser()` or `FhirParserFactory.createR4Parser()`.
+
+Decoding and Encoding JSON FHIR data is as simple as:
+
+==== Decode
+
+In case you know the type of FHIR JSON data, you could directly decode it.
+
+[source, Kotlin]
 ----
+val parser = FhirParserFactory.createR4Parser()
 
-Make sure you are in the project root and install the correct Python version which is derived from `.python-version` file:
+val data = "JSON FHIR data"
 
-[source,bash]
+val result = parser.fromJson(DomainResource::class, "JSON FHIR data")
 ----
-pyenv install
+
+==== Encode
+
+Just pass your FHIR data to the corresponding FHIR parser to encode to JSON format.
+
+[source, Kotlin]
 ----
+val parser = FhirParserFactory.createR4Parser()
 
-==== Install Fastlane
+val data = DomainResource()
 
-Have a look at link:fastlane/Readme.md[Readme]
+val result = parser.toJson(data)
 
-==== Android Studio
+----
 
-For better interop with iOS you may install the link:https://plugins.jetbrains.com/plugin/14936-kotlin-multiplatform-mobile[KMM plugin for Android Studio]]. It let's you run, test and debug shared code on Android and iOS without switching the IDEs.
+=== Model
 
-== Usage
+FHIR Resources and Elements are represented as Kotlin interfaces which are named `Fhir` + ResourceType` e.g. `FhirDocumentReference`. While a `FhirDocumentReference` is implemented by `DocumentReference` which is a Kotlin data class. So they are equatable, copyable, destructable by default.
 
-=== First time
+[source, Kotlin]
+----
+val name = HumanName(family = "Doe", given = listOf("John"))
+val patient = Patient(name = listOf(name))
+
+val newName = HumanName(family = "Johnny", given = listOf("Silverhand"))
+val patientCopy = patient.copy(name = listOf(newName))
```

Comment:
Could we use maybe a non-binary name like `Andi` or at least one female name here?